src/vmm/src/builder.rs

```diff
@@ -550,11 +558,21 @@ fn load_kernel(
         .try_clone()
         .map_err(|e| StartMicrovmError::Internal(Error::KernelFile(e)))?;
 
-    let entry_addr =
-        kernel::loader::load_kernel(guest_memory, &mut kernel_file, arch::get_kernel_start())
-            .map_err(StartMicrovmError::KernelLoader)?;
+    let entry_addr = Loader::load::<std::fs::File, GuestMemoryMmap>(
+        guest_memory,
+        #[cfg(target_arch = "x86_64")]
+        None,
+        #[cfg(target_arch = "aarch64")]
+        Some(GuestAddress(arch::get_kernel_start())),
+        &mut kernel_file,
+        #[cfg(target_arch = "x86_64")]
+        Some(GuestAddress(arch::get_kernel_start())),
+        #[cfg(target_arch = "aarch64")]
+        None,
```

Comment:
these inline cfg's are hard to parse visually. Maybe add conditional compilation to the entire variable definition?